spec/queries/keywords_query_spec.rb

```diff
@@ -4,36 +4,177 @@
+        
+        context 'when there are urls containing the word vpn' do
+          it 'returns 2 keywords' do
+            user = Fabricate(:user)
+            ads_top_urls = ['https://www.thetopvpn.com', 'https://www.nordvpn.com', 'https://www.vnexpress.net']
+            Fabricate.times(2, :keyword, ads_top_urls: ads_top_urls, user: user)
+
+            filter_params = { adwords_url_contains: 'vpn' }
+            keywords = described_class.new(user.keywords, filter_params).call
+
+            expect(keywords.count).to eq 2
+          end
+        end
+
+        context 'when there is no urls containing the word vpn' do
+          it 'returns empty array for keywords' do
+            user = Fabricate(:user)
+            ads_top_urls = ['https://www.thetop.com', 'https://www.nord.com', 'https://www.vnexpress.net']
+            Fabricate(:keyword, ads_top_urls: ads_top_urls, user: user)
 
-          expect(keywords.count).to eq 2
+            filter_params = { adwords_url_contains: 'vpn' }
+            keywords = described_class.new(user.keywords, filter_params).call
+
+            expect(keywords).to be_empty
+          end
         end
       end
+    end
 
-      context 'given ads_top_urls does not contain the filtering word apple' do
+    context 'when querying with result_url_contains' do
+      context 'given user does not have keyword' do
         it 'returns an empty array' do
+          Fabricate(:keyword)
           user = Fabricate(:user)
-          ads_top_urls = ['https://www.thetopvpn.com', 'https://www.nordvpn.com', 'https://www.vnexpress.net']
-          Fabricate.times(2, :keyword, ads_top_urls: ads_top_urls, user: user)
-          filter_params = { adwords_url_contains: 'apple' }
-          keywords = described_class.new(Keyword, filter_params).call
+
+          filter_params = { result_url_contains: 'vpn' }
+          keywords = described_class.new(user.keywords, filter_params).call
 
           expect(keywords).to be_empty
         end
       end
 
-      context 'given filter_params is nil' do
-        it 'returns Keyword scope' do
-          keywords = described_class.new(Keyword, nil).call
+      context 'when user has keyword' do
+        context 'when there are search urls containing the word vpn' do
+          it 'returns 2 keywords' do
+            user = Fabricate(:user)
+            result_urls = ['https://www.thetopvpn.com', 'https://www.nordvpn.com', 'https://www.vnexpress.net']
+            Fabricate.times(2, :keyword, result_urls: result_urls, user: user)
 
-          expect(keywords).to eq Keyword
+            filter_params = { result_url_contains: 'vpn' }
+            keywords = described_class.new(user.keywords, filter_params).call
+
+            expect(keywords.count).to eq 2
+          end
+        end
+
+        context 'when there is no urls containing the word vpn' do
+          it 'returns empty array for keywords' do
+            user = Fabricate(:user)
+            Fabricate(:keyword, user: user)
```

Comment:
Actually, the user play no role in the query object, querying with user's id condition is added to the query by Pundit in the `KeywordController`, I also removed `user: user` in my PR